libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -889,15 +903,22 @@ void AP_GPS::update_primary(void)
         return;
     }
 
-    if ((GPSAutoSwitch)_auto_switch.get() == GPSAutoSwitch::NONE) {
-        // AUTO_SWITCH is 0 so no switching of GPSs, always use first instance
-        primary_instance = 0;
-        return;
+    // check the primary param is set to possible GPS
+    int8_t primary_param = _primary.get();
+    if ((primary_param < 0) || (primary_param>=GPS_MAX_RECEIVERS)) {
+        GCS_SEND_TEXT(MAV_SEVERITY_INFO, "GPS_PRIMARY %i out of range", primary_param);
+        _primary.set_and_save(0);
```

Comment:
Not sure if we should be status texting here and saving the parameter. We don't normally just force the actual parameter value on storage, we may force the current value, but I'm not sure set and save is correct here.

Reply:
removed